### PR TITLE
fix: enable to use s3 compatible

### PR DESCRIPTION
**Pull request by tk3fftk**
The helper client only points to AWS.
This PR enables to use cache invalidation in S3 compatible.
And add more logs for debugging.

### CODE REVIEW

File: plugins/caches.js [hunk 1]
```diff
@@ -404,14 +408,18 @@ exports.plugin = {
...
-                    .catch((err) => {
-                        if (err === 'Permission denied') {
-                            return boom.forbidden(err);
-                        }
+                })).then(() => {
+                    request.log([cachePath, 'info'], 'Successfully deleted a cache');
 
-                        return h.response().code(500);
-                    });
+                    return h.response().code(200);
```
Comment (yuichi10) [inline plugins/caches.js]: [imo] Is it better to use status code 204, because success of delete?

File: plugins/commands.js [hunk 2]
```diff
@@ -140,10 +140,13 @@ exports.plugin = {
 
                 try {
                     await cache.drop(id);
+                    request.log([id, 'info'], 'Successfully deleted a command');
 
-                    return h.response();
+                    return h.response().code(200);
```
Comment (yuichi10) [inline plugins/commands.js]: Same here.

File: plugins/caches.js (hunk 1 above)
Comment (tk3fftk) [reply plugins/caches.js]: 204 is reasonable. fixed.
https://stackoverflow.com/a/29574521

**Comment by tk3fftk** [comment]
Test stack may cause updating `jenkins-mocha`. I try to fix it.

**Comment by tk3fftk** [comment]
Added `--exit` same as api.
 https://github.com/screwdriver-cd/screwdriver/blob/4ec4b9acf0545658a97369567ffdb421fea767de/package.json#L8

It's mocha v4 behavior.
https://boneskull.com/mocha-v4-nears-release

**Pull request merged**